app/models/shared_ai_conversation.rb

```diff
@@ -0,0 +1,201 @@
+# frozen_string_literal: true
+
+class SharedAiConversation < ActiveRecord::Base
+  DEFAULT_MAX_POSTS = 100
+
+  belongs_to :user
+  belongs_to :target, polymorphic: true
+
+  validates :user_id, presence: true
+  validates :target, presence: true
+  validates :context, presence: true
+  validates :share_key, presence: true, uniqueness: true
+
+  before_validation :generate_share_key, on: :create
+
+  def self.share_conversation(user, target, max_posts: DEFAULT_MAX_POSTS)
+    raise "Target must be a topic for now" if !target.is_a?(Topic)
+
+    conversation = find_by(user: user, target: target)
+    conversation_data = build_conversation_data(target, max_posts: max_posts)
+
+    if conversation
+      conversation.update(**conversation_data)
+      conversation
+    else
+      create(user_id: user.id, target: target, **conversation_data)
+    end
+  end
+
+  # Technically this may end up being a chat message
+  # but this name works
+  class SharedPost
+    attr_accessor :user
+    attr_reader :id, :user_id, :created_at, :cooked, :persona
+    def initialize(post)
+      @id = post[:id]
+      @user_id = post[:user_id]
+      @created_at = DateTime.parse(post[:created_at])
+      @cooked = post[:cooked]
+      @persona = post[:persona]
+    end
+  end
+
+  def populated_context
+    return @populated_context if @populated_context
+    @populated_context = context.map { |post| SharedPost.new(post.symbolize_keys) }
+    populate_user_info!(@populated_context)
+    @populated_context
+  end
+
+  def to_json
+    posts =
+      self.populated_context.map do |post|
+        {
+          id: post.id,
+          cooked: post.cooked,
+          username: post.user.username,
+          created_at: post.created_at,
+        }
+      end
+    { llm_name: self.llm_name, share_key: self.share_key, title: self.title, posts: posts }
+  end
+
+  def url
+    "#{Discourse.base_uri}/discourse-ai/ai-bot/shared-ai-conversations/#{share_key}"
+  end
+
+  def html_excerpt
+    html = +""
+    populated_context.each do |post|
+      text =
+        PrettyText.excerpt(
```

Comment:
You can just do `post.excerpt` here too like you do lower in `self.excerpt`; it accepts the same arguments as using PrettyText.excerpt directly:

https://github.com/discourse/discourse/blob/6435b790abc932cb99ceb81ceca7df0d15186f7d/app/models/post.rb#L521-L524

Reply:
diff post object :)